cmd/ndm_daemonset/controller/devicestore.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	"github.com/openebs/node-disk-manager/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateDevice creates the Device resource in etcd
+func (c *Controller) CreateDevice(dvr apis.Device) {
+
+	dvrCopy := dvr.DeepCopy()
+	err := c.Clientset.Create(context.TODO(), dvrCopy)
+	if err == nil {
+		glog.Info("Created device object in etcd : ",
+			dvrCopy.ObjectMeta.Name)
+		return
+	}
+	glog.Infof("Dvr:%#v", dvrCopy)
+	glog.Error("Creation of device object failed : ", err)
+
+	/*
+	 * Creation may fail because resource is already exist in etcd.
```

Comment:
Will it be good to check for specific error & continue for update ?

Reply:
Added code to check error and take a call to return or update.